### PR TITLE
Move the switch to classic shortcode block button to separate component

### PR DESCRIPTION
**Pull request by tarunvijwani**
### Changes in the PR
- Move modal component file from `incompatible-extension-notice` component folder to `switch-to-classic-shortcode-button` folder.
- Rename modal component file to model-content component file.
- Move the switch to classic shortcode block button to separate component folder

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Related #42072 

We're updating the generic incompatibility notice content and adding the switch to classic block button to it. 
In this PR, we're only moving the switch to classic shortcode block button to separate component so that it can reused and added to the generic incompatibility notice.

PR: #42751 will take care of the updating the generic incompatibility notice content and adding the switch to classic block button to it.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install any incompatible plugin like PayTm
2. Edit the Cart block page.
3. Focus on the Cart block. See the incompatibility notice in the sidebar. Click the "switch to classic cart" button.
4. Modal should open. Click switch. Block should be replaced.
5. In the snackbar bottom left, click "undo". Should revert back to blocks.
6. Repeat the steps for the Checkout Block page editor. 

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 

Move the switch to classic shortcode block button to separate component.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/editor-components/switch-to-classic-shortcode-button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+												: 'cart',
+									}
+								);
+								closeModal();
+							} }
+						>
+							{ __( 'Cancel', 'woocommerce' ) }
+						</Button>
+					</TabbableContainer>
+				</Modal>
```
Comment (nielslange) [inline plugins/woocommerce-blocks/assets/js/editor-components/switch-to-classic-shortcode-button/index.tsx:142]: This block is pretty long, which affects the readability. What are your thoughts on splitting it into smaller parts? 

> [!IMPORTANT]
> The code snippets below were generated using ChatGPT-4 to showcase potential enhancements in the readability of the <ModalContent> section. Please note that they have not been thoroughly reviewed and are not intended for direct implementation without further validation.

The main function at the end of the file could look like this:

```ts
<ModalContent blockType={block} />
    <TabbableContainer className="wc-blocks-switch-to-classic-shortcode-modal-actions">
        <SwitchButton />
        {' '}
        <CancelButton />
    </TabbableContainer>
</Modal>
```

Here, we're simply calling the components `<SwitchButton>` and `<CancelButton />`. These components could look like this:

```ts
const SwitchButton = () => (
    <Button variant="primary" isDestructive={true} onClick={handleSwitchClick}>
        { __( 'Switch', 'woocommerce' ) }
    </Button>
);

const CancelButton = () => (
    <Button variant="secondary" onClick={handleCancelClick}>
        { __( 'Cancel', 'woocommerce' ) }
    </Button>
);
```

Finally, we'd define functions for the `onClick` events. These functions could look like this:

```ts
const handleSwitchClick = () => {
    replaceBlock(
        clientId,
        createBlock('woocommerce/classic-shortcode', {
            shortcode: block === 'woocommerce/checkout' ? 'checkout' : 'cart',
        })
    );
    recordEvent('switch_to_classic_shortcode_confirm', {
        shortcode: block === 'woocommerce/checkout' ? 'checkout' : 'cart',
    });
    selectClassicShortcodeBlock();
    createInfoNotice(snackbarLabel, {
        actions: [
            {
                label: __( 'Undo', 'woocommerce' ),
                onClick: handleUndoClick,
            },
        ],
        type: 'snackbar',
    });
    closeModal();
};

const handleUndoClick = () => {
    undo();
    recordEvent('switch_to_classic_shortcode_undo', {
        shortcode: block === 'woocommerce/checkout' ? 'checkout' : 'cart',
    });
};

const handleCancelClick = () => {
    recordEvent('switch_to_classic_shortcode_cancel', {
        shortcode: block === 'woocommerce/checkout' ? 'checkout' : 'cart',
    });
    closeModal();
};
```

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @tarunvijwani. So far, I've only looked at the code and left some suggestions. Please let me know if you have any questions. If I should review the PR again, just re-request another review from me. 😀

File: plugins/woocommerce-blocks/assets/js/editor-components/switch-to-classic-shortcode-button/editor.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	li {
+		list-style: none outside;
+		margin: 1em 0;
+		padding: 0;
+	}
+	li::before {
+		content: "❌";
+		margin-right: 0.5em;
+	}
+}
```
Comment (nielslange) [inline plugins/woocommerce-blocks/assets/js/editor-components/switch-to-classic-shortcode-button/editor.scss]: As we're using SCSS, we can change `li::before` to `&::before` and nest it within the `<li>` element. I'd also add line breaks. Here's my suggestion:

```suggestion
ul.cross-list {
  list-style: none outside;
  margin: 0 0 24px 1em;
  padding: 0;

  li {
    list-style: none outside;
    margin: 1em 0;
    padding: 0;

    &::before {
      content: "❌";
      margin-right: 0.5em;
    }
  }
}
```

**Comment by tarunvijwani** [comment]
Thank you for your review and excellent suggestions, @nielslange! I have made changes to improve the readability of the component by moving out the functions. Instead of creating new components such as `SwitchButton` or `CancelButton`, I kept the existing buttons components to avoid redundant code.

**Review by nielslange — APPROVED** [review]
Thanks for applying the suggestions, @tarunvijwani. 

> Instead of creating new components such as `SwitchButton` or `CancelButton`, I kept the existing buttons components to avoid redundant code.

That sounds good. Too much abstraction isn't necessary better. 

I've successfully tested this PR. Let's ⛴️ this improvement. 🙌

**Pull request merged**